modMcf/src/org/aion/mcf/db/TransformedCodeInfo.java

```diff
@@ -0,0 +1,43 @@
+package org.aion.mcf.db;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.aion.util.types.ByteArrayWrapper;
+
+public class TransformedCodeInfo {
+
+    // Key for this map is the codeHash, the value is a map which has AVM version as its key and transformed code as its value.
+    public Map<ByteArrayWrapper, Map<Integer, ByteArrayWrapper>> transformedCodeMap;
+
+    public TransformedCodeInfo() {
+        transformedCodeMap = new HashMap<>();
+    }
+
+    public void add(ByteArrayWrapper codeHash, int avmVersion, ByteArrayWrapper transformedCode) {
+        Map<Integer, ByteArrayWrapper> versionToTransformedCode = transformedCodeMap.get(codeHash);
+
+        if (versionToTransformedCode == null) {
+            versionToTransformedCode = new HashMap<>();
+        }
+        versionToTransformedCode.put(avmVersion, transformedCode);
+        transformedCodeMap.put(codeHash, versionToTransformedCode);
+    }
+
+    public byte[] getTransformedCode(ByteArrayWrapper codeHash, int avmVersion) {
+        Map<Integer, ByteArrayWrapper> avmVersionMap = transformedCodeMap.get(codeHash);
+
+        if (avmVersionMap == null) return null;
+
+        ByteArrayWrapper transformedCode = avmVersionMap.get(avmVersion);
+
+        return transformedCode == null ? null : transformedCode.toBytes();
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
When implementing `equals` we also need `hashCode`. Where is this used? If possible I would not implement it at all and rely on object equality.